factoryWithTypeCheckers.js

```diff
@@ -314,6 +315,36 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
     return createChainableTypeChecker(validate);
   }
 
+
+  function createElementOfTypeChecker(expectedTypes) {
+    var ACCEPTABLE_TYPES_OF_EXPECTED_TYPES = ['string', 'function'];
+    function getDisplayName (comp) {
+      return typeof comp === 'function' ? comp.displayName || comp.name : comp;
+    }
+
+    function validate(props, propName, componentName, location, propFullName) {
+      expectedTypes = [].concat(expectedTypes)
```

Comment:
This should throw if `expectedTypes` isn't an array (if it takes an array), like `oneOf`/`oneOfType` do.

Reply:
i'm hoping to give user some flexibility in case they only want to pass a single expected element type.

Reply:
Flexibility in APIs leads to bugs; if they want to pass a single type, they can add `[ ]` without difficulty.